### PR TITLE
Implement test coverage report - Closes #408

### LINKED ISSUE #408: Add test coverage metric
Use tool for test coverage. Integrate it with a tool for test coverage display. Should be also possible to see aggregated test coverage metrics gathered from coverages from all parallelized tests run by Travis.

### PR DESCRIPTION
**Pull request by MaciejBaj**
Travis runs tests one by one. After the job is successfully finished, depending on type (unit / functional) proper test coverage report is gathered (functional test coverage must be gathered from the server, where the tests had been running). 

Submitting multiple chunked jobs and merging them into one is done using coveralls webhook mechanism (https://coveralls.zendesk.com/hc/en-us/articles/203484329-Parallel-Build-Webhook), so no external server for reports storage is needed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,7 @@ Lisk is a next generation crypto-currency and decentralized application platform
 
 [![Join the chat at https://gitter.im/LiskHQ/lisk](https://badges.gitter.im/LiskHQ/lisk.svg)](https://gitter.im/LiskHQ/lisk?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
 [![Build Status](https://travis-ci.org/LiskHQ/lisk.svg?branch=development)](https://travis-ci.org/LiskHQ/lisk)
+[![Coverage Status](https://coveralls.io/repos/github/MaciejBaj/lisk/badge.svg?branch=development)](https://coveralls.io/github/MaciejBaj/lisk?branch=development)
```
Comment (karmacoma) [inline README.md]: Please change to LiskHQ.

File: app.js [hunk 2]
```diff
@@ -34,6 +34,7 @@ program
 	.option('-y, --sync [peers...]', 'peers list to sync with')
 	.option('-l, --log <level>', 'log level')
 	.option('-s, --snapshot <round>', 'verify snapshot')
+	.option('-t, --coverage', 'enable functional tests code coverage')
```
Comment (karmacoma) [inline app.js]: Only functional tests?

If not: `enable functional tests code coverage` > `measure test code coverage`

If yes: `enable functional tests code coverage` > `measure functional test code coverage`

File: package.json [hunk 3]
```diff
@@ -49,9 +50,9 @@
   "devDependencies": {
     "bitcore-mnemonic": "=1.1.1",
     "browserify-bignum": "=1.3.0-2",
-    "buffer": "=4.9.1",
     "chai": "=3.5.0",
     "chai-bignumber": "=2.0.0",
+    "coveralls": "^2.11.16",
```
Comment (karmacoma) [inline package.json]: Please fixate dependency version.

File: app.js [hunk 4]
```diff
@@ -168,6 +173,13 @@ d.run(function () {
 			var cors = require('cors');
 			var app = express();
 
+			if (appConfig.coverage) {
+				var im = require('istanbul-middleware');
+				logger.debug('Hook loader for coverage - ensure this is not production!');
```
Comment (karmacoma) [inline app.js]: `ensure this is not production!` > `do not use in production environment!`

What are actual implications here?

File: app.js [hunk 5]
```diff
@@ -75,6 +76,10 @@ if (program.snapshot) {
 	);
 }
 
+if (program.coverage) {
+	appConfig.coverage = true;
+}
+
```
Comment (karmacoma) [inline app.js]: I don't mind this approach tbh, but maybe better to enable through: `NODE_ENV=test` environment variable?

File: .travis.yml [hunk 6]
```diff
@@ -22,31 +22,40 @@ before_script:
   - tar -zxvf lisk-node-Linux-x86_64.tar.gz
   - cd test/lisk-js/; yarn; cd ../..
   - cp test/config.json test/genesisBlock.json .
-  - node app.js &> .app.log &
+  - node app.js --coverage=true &> .app.log &
+notifications:
+  webhooks: https://coveralls.io/webhook?repo_token=7s05KDqmPWkwZ6nzU5WtznKkt5FKDE3kv
```
Comment (karmacoma) [inline .travis.yml]: Do we need to change this token? i.e. for LiskHQ.

Comment (MaciejBaj) [inline .travis.yml]: exactly, sent you the link

File: package.json (hunk 3 above)
Comment (MaciejBaj) [inline package.json]: sry

File: app.js (hunk 2 above)
Comment (MaciejBaj) [inline app.js]: yes, only for functional. Unit tests run on their own 'instance' of lisk and they store the coverage straight in the directory: `test/.coverage-unit`. The purpose of using `istanbul-middleware` plugin is to store the code coverage on the test server, which is requested via API calls by the functional tests.

File: app.js (hunk 4 above)
Comment (MaciejBaj) [inline app.js]: running coverage on production will just gather code coverage on production, which just adds some extra work for the nodes.

**Pull request merged**